src/logic/tokens/utils/tokenHelpers.ts

```diff
@@ -45,10 +48,17 @@ export const isTokenTransfer = (tx: TxServiceModel): boolean => {
 export const isSendERC721Transaction = (tx: TxServiceModel, txCode?: string, knownTokens?: TokenState): boolean => {
   // "0x57f1887a8BF19b14fC0dF6Fd9B2acc9Af147eA85" - ens token contract, includes safeTransferFrom
   // but no proper ERC721 standard implemented
+  const ENS_TOKEN_CONTRACT = '0x57f1887a8BF19b14fC0dF6Fd9B2acc9Af147eA85'
+  let hasERC721Transfer = false
+
+  if (tx.dataDecoded && sameString(tx.dataDecoded.method, SAFE_METHODS_NAMES.SAFE_TRANSFER_FROM)) {
+    hasERC721Transfer = tx.dataDecoded.parameters.findIndex((param) => sameString(param.name, 'tokenId')) !== -1
+  }
+
   return (
-    (txCode?.includes(SAFE_TRANSFER_FROM_WITHOUT_DATA_HASH) &&
-      tx.to !== '0x57f1887a8BF19b14fC0dF6Fd9B2acc9Af147eA85') ||
-    (isTokenTransfer(tx) && !knownTokens?.get(tx.to))
+    (txCode?.includes(SAFE_TRANSFER_FROM_WITHOUT_DATA_HASH) && !sameAddress(tx.to, ENS_TOKEN_CONTRACT)) ||
+    (isTokenTransfer(tx) && !knownTokens?.get(tx.to)) ||
+    hasERC721Transfer
```

Comment:
I'm not sure how reliable this `isSendERC721Transaction` function is.

Is it safe to just rely on what the tx-service returns? If so, maybe we can just verify that `hasERC721Transfer` @dasanra, any opinion?

Reply:
I refactored this could you check this again?